src/main/java/xyz/aadev/generitech/common/tileentities/machines/TileEntityFurnace.java

```diff
@@ -162,22 +169,35 @@ public float getTempRate() {
 
     @Override
     public void update() {
-        if (machineActive & !isSmeltPaused) {
+        if (container.getStoredPower()>=powerUsage && machineActive & !isSmeltPaused) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Correct this "&" to "&&". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS2178)